# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/src/proc/terminator.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/naga/src/proc/terminator.rs b/third_party/rust/naga/src/proc/terminator.rs
--- a/third_party/rust/naga/src/proc/terminator.rs
+++ b/third_party/rust/naga/src/proc/terminator.rs
@@ -34,12 +34,13 @@ pub fn ensure_block_returns(block: &mut 
         | Some(&mut S::Break)
         | Some(&mut S::Continue)
         | Some(&mut S::Return { .. })
         | Some(&mut S::Kill) => (),
         Some(&mut S::Loop { .. })
         | Some(&mut S::Store { .. })
         | Some(&mut S::ImageStore { .. })
         | Some(&mut S::Call { .. })
+        | Some(&mut S::Atomic { .. })
         | Some(&mut S::Barrier(_))
-        | None => block.push(S::Return { value: None }),
+        | None => block.push(S::Return { value: None }, Default::default()),
     }
 }